benchmarks/tosi_et_al_2015_gcubed/tosi.cc

```diff
@@ -0,0 +1,621 @@
+                statistics.set_scientific ("Rate of work (W)", true);
+                statistics.add_value ("Error", error);
+                statistics.set_precision ("Error", 8);
+                statistics.set_scientific ("Error", true);
+              }
+
+            std::ostringstream output;
+            output.precision(3);
+            if (this->convert_output_to_years() == true)
+              output << viscous_dissipation *year_in_seconds
+                     << " J/yr "
+                     << work *year_in_seconds
+                     << " J/yr "
+                     << error
+                     << " error";
+            else
+              output << viscous_dissipation
+                     << " W "
+                     << work
+                     << " W "
+                     << error
+                     << " error";
+
+            return std::pair<std::string, std::string> ("Viscous dissipation, rate of work, error:",
+                                                        output.str());
+    }
+
+
+  }
+}
+
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace TosiBenchmark
+  {
+    ASPECT_REGISTER_MATERIAL_MODEL(TosiMaterial,
+        "TosiMaterial",
+        "A material model that has constant values "
+        "for all coefficients but the density and viscosity as described in the "
+        "open access paper of Tosi et al. 2015. "
+        "The default parameter values are chosen according to Case 1 of the paper. "
+        "All of the values that define this model are read "
+        "from a section ``Material model/Tosi model'' in the input file, see "
+        "Section~\\ref{parameters:Material_model/Tosi_model}."
+        "\n\n"
+        "This model uses the following set of equations for the two coefficients that "
+        "are non-constant (see equation (6) - (10) of the paper): "
+        "\\begin{align}"
+        "  \\eta(T,z,\\dot \\epsilon) &= 2\\frac{1}{\\frac{1}{\\eta_{lin}(T,z)}\\frac{1}{\\eta_{plast}(\\dot\\epsilon}}, \\\\"
+        "  \\rho(T) &= \\left(1-\\alpha (T-T_0)\\right)\\rho_0,"
+        "\\end{align}"
+        "where z represents depth."
+        "\n\n"
+        "The linear and plastic viscosity parts are defined as follows:"
+        "\\begin{align}"
+        "  \\eta_{lin}(T,z) &= \\exp(-\\ln(\\eta_T)T+\\ln(\\eta_z)z), \\\\"
+        "  \\eta_{plast}(\\dot\\epsilon) &= \\eta^{*}+\\frac{\\sigma_y}{\\sqrt(\\dot\\epsilon : \\dot\\epsilon)} "
+        "\\end{align} "
+        "\n\n"
+        "Note that this model uses the formulation that assumes an incompressible "
+        "medium despite the fact that the density follows the law "
+        "$\\rho(T)=\\rho_0(1-\\alpha(T-T_0))$. ")
+    ASPECT_REGISTER_POSTPROCESSOR(TosiPostprocessor,
+                    "TosiPostprocessor",
+                    "A postprocessor that computes the viscous dissipation"
+                    "for the whole domain as: "
+                    "$<\\Phi>=\\int_{V} \\tau : \\dot{\\epsilon}dV$ "
```

Comment:
not important here, but in general: `$<...>$` often comes out looking pretty awkward because these are less-than and greater-than signs with a small opening angle. It looks a lot better if you say `$\left<...\right>$` because that's now a set of brackets with a large opening angle.